knime_extension/src/nodes/spatialstatistics.py

```diff
@@ -115,42 +115,58 @@ class spatialWeights:
             "Rook",
             "Binary Distance Band",
             "Inverse Distance",
-            "Lattice",
             "K nearest",
             "Lattice",
             "Kernel",
             "Get spatial weights matrix from file",
         ],
     )
     order = knext.IntParameter(
-        "Order for Queen or Rook",
+        "Order",
         """The order of the weight matrix is 1 by default. Users can change the order of the weights, higher order 
         weights will treat further units as neighbors.""",
         1,
+    ).rule(
+        knext.OneOf(
+            category,
+            [
+                "Queen",
+                "Rook",
+                "Binary Distance Band",
+                "Inverse Distance",
+                "K nearest",
+                "Lattice",
+            ],
+        ),
+        knext.Effect.SHOW,
     )
 
     Threshold = knext.IntParameter(
         "Threshold for Inverse Distance or Binary Distance Band",
         """The distance threshold for constructing binary distance band and inverse distance weights. Defaults to 1""",
         1,
+    ).rule(
+        knext.OneOf(category, ["Binary Distance Band", "Inverse Distance"]),
+        knext.Effect.SHOW,
     )
 
-    Power = knext.IntParameter(
-        "Power for Inverse Distance",
-        """The power for constructing inverse distance weights. Defaults to 1.""",
-        1,
-    )
+    # Power = knext.IntParameter(
```

Comment:
Why did you comment this out? Is this change backward compatible?

Reply:
We don't use this parameter in the spatial weight